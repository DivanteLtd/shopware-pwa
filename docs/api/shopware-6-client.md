<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@shopware-pwa/shopware-6-client](./shopware-6-client.md)

## shopware-6-client package

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [ClientSettings](./shopware-6-client.clientsettings.md) | <b><i>(BETA)</i></b> |

## Variables

|  Variable | Description |
|  --- | --- |
|  [config](./shopware-6-client.config.md) | <b><i>(BETA)</i></b> |

