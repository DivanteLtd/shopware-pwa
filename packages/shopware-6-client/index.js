"use strict";

if (process.env.NODE_ENV === "production") {
  module.exports = require("./dist/shopware-6-client.cjs.prod.js");
} else {
  module.exports = require("./dist/shopware-6-client.cjs.js");
}
