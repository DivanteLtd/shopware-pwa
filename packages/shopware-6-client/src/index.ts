export * from "./index.inner";

// Explicitly import offline overrides and alias them

// Product Listing
import { getCategoryProducts as offlineGetCategoryProducts } from "./offline-services/productService";

const getCategoryProducts = offlineGetCategoryProducts;

// Product Search
import {
  searchProducts as offlineSearchProducts,
  searchSuggestedProducts as offlineSearchSuggestedProducts,
} from "./offline-services/searchService";

const searchProducts = offlineSearchProducts;
const searchSuggestedProducts = offlineSearchSuggestedProducts;

// Re-export them with original name, because local modules have priority in bundler
export { searchProducts, searchSuggestedProducts, getCategoryProducts };
