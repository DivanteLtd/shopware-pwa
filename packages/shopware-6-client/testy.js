const { getUserCountry, setup } = require("@shopware-pwa/shopware-6-client");

!(async () => {
  setup({
    endpoint: "http://localhost:8000",
    accessToken: "SWSCMGPZM1LVCK5QOHLBSMLAUQ",
  });
  try {
    const response = await getUserCountry("012e32ef96674dd584d71d993fb1f458");
    console.info(response);
  } catch (error) {
    console.warn("error", error);
  }
})();
