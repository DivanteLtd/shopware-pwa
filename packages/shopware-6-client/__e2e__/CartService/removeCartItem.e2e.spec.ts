import {addProductToCart, removeCartItem} from "@shopware-pwa/shopware-6-client";
describe("shopware-6-client - E2E - CartService - addCartItemQuantity", () => {
  it("should test remove product from cart response", async() => {
    await addProductToCart("044a190a54ab4f06803909c3ee8063ef", 2);
    const result = await removeCartItem("044a190a54ab4f06803909c3ee8063ef")
    expect(result).toMatchSnapshot()
  });

  it("should test remove product from cart error reponse", async() => {
    try {
      await removeCartItem("044a190a54ab4f06803909c3ee8063ef")
      expect("didn't throw an error").toEqual("should throw an error");
    } catch(e) {
      expect(e).toMatchSnapshot();
    }
  });
});