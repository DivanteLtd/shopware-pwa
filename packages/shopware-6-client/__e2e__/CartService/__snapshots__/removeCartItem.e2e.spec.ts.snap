// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CartService - addCartItemQuantity should test remove product from cart error reponse 1`] = `[Error: Request failed with status code 400]`;
